patches/server/1045-Fix-PickupRule-reset-when-deflected.patch

```diff
@@ -0,0 +1,55 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Tamion <70228790+notTamion@users.noreply.github.com>
+Date: Sun, 21 Jul 2024 19:11:22 +0200
+Subject: [PATCH] Fix PickupRule reset when deflected
+
+
+diff --git a/src/main/java/net/minecraft/world/entity/projectile/AbstractArrow.java b/src/main/java/net/minecraft/world/entity/projectile/AbstractArrow.java
+index ddf47dab1ab92c45e3eea09239d418a9798ed59e..b9cce1f10fbc4b1dd017c63a0474639f37145d2e 100644
+--- a/src/main/java/net/minecraft/world/entity/projectile/AbstractArrow.java
++++ b/src/main/java/net/minecraft/world/entity/projectile/AbstractArrow.java
+@@ -666,9 +666,9 @@ public abstract class AbstractArrow extends Projectile {
+ 
+     }
+ 
+-    @Override
+-    public void setOwner(@Nullable Entity entity) {
++    public void setOwner(@Nullable Entity entity, boolean updatePickup) { // Paper - Fix PickupRule reset when deflected
+         super.setOwner(entity);
++        if (!updatePickup) return; // Paper - Fix PickupRule reset when deflected
+         Entity entity1 = entity;
+         byte b0 = 0;
+ 
+@@ -702,6 +702,13 @@ public abstract class AbstractArrow extends Projectile {
+         this.pickup = entityarrow_pickupstatus;
+     }
+ 
++    // Paper start - Fix PickupRule reset when deflected
```

Comment:
This diff can be moved into the other diff.
E.g.
```java
@Override
public void setOwner(Entity) {
// Paper start - ...
setOwner(entity, true);
}

public void setOwner(Entity, boolean) {
  super.setOwner(entity);
  if (!updatePickup) return;
// Paper end - ...
  ..
}

Reply:
so we don't have as many paper comments?

Reply:
Just keeps the diff nicely together in the class.